aws_embedded_metrics/serializers/log_serializer.py

```diff
@@ -53,4 +55,13 @@ def serialize(context: MetricsContext) -> str:
 
             metric_pointers.append({"Name": metric_name, "Unit": metric.unit})
 
-        return json.dumps(body)
+            should_serialize: bool = len(metric_pointers) == MAX_METRICS_PER_EVENT
+            if should_serialize:
+                event_batches.append(json.dumps(body))
+                metric_pointers = []
+                body["_aws"]["CloudWatchMetrics"][0]["Metrics"] = metric_pointers
```

Comment:
do we ever reset the body? If I'm reading this correctly, all JSON payloads will contain all the metric values just not the pointers? For example, in the node implementation, we recreate a new body every time we flush: https://github.com/awslabs/aws-embedded-metrics-node/blob/730bbaeb341ee3952a404c04fc1de59a6ec65cf5/src/serializers/LogSerializer.ts#L71

Reply:
You are correct, JSON payload will contain the metric values. As per my understanding, We need to reset Metrics ( body["_aws"]["CloudWatchMetrics"][0]["Metrics"] ) and use the same value for other fields - Dimensions, Namespace, etc.  Please correct me if am wrong.

Thanks,
Abdul

Reply:
We need to reset the body as well to prevent the payload from continuing to grow with metrics that aren't referenced. You're removing it from the pointers which is good, but not from the body which get added into ln54.

Reply:
Thanks for pointing it out. Fixed.